core/src/checkpointing/hdf5/combined.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <Python.h> // has to be the first included header
+
+#include "control/dihu_context.h"
+#include "checkpointing/generic.h"
+#include "output_writer/hdf5/hdf5.h"
+
+namespace Checkpointing {
+namespace HDF5 {
+//! The concreate implementation for HDF5 combined checkpointing
+class Combined : public Generic {
+public:
+  //! Constructs a new combined HDF5 checkpointing implementation
+  Combined(DihuContext context,
+           std::shared_ptr<Partition::RankSubset> rankSubset = nullptr,
+           const std::string &prefix = ".");
+
+  //! Create a new checkpoint with the problem data as well as the timestep and
+  //! the current time.
+  template <typename DataType>
+  void createCheckpoint(DataType &problemData, int timeStepNo = -1,
```

Comment:
Is there any special reason to assign rvalues in the function signature ? This is not a function override from the Generic base class.